db.js

```diff
@@ -262,6 +263,16 @@ exports.init = function (dir, config) {
     })
   }
 
+  async function close(cb) {
```

Comment:
promisify used here so we can run all these child `close` in parallel and use Promise.all to collect results. writing the equivalent with pure callbacks is a mind twist

Reply:
wouldn't you be able to not have to use async on the function if you resolved the promise.all to the cb? Basically that is the only async call there.

Reply:
oh yeah, that's another way of doing it

```js
  function close(cb) {
    const tasks = []
    tasks.push(promisify(log.close)())
    for (const indexName in indexes) {
      tasks.push(promisify(indexes[indexName].close)())
    }
    Promise.all(tasks).then(cb)
  }
```

the difference is that `async function` would allow using it as either `await db.close()` or `db.close(() => { ..... })`, the normal function wouldn't.

i'm fine either way

Reply:
actually, `return Promise.all(tasks).then(cb)` then both usage styles are enabled!

Reply:
Updated PR